src/main/java/com/google/devtools/build/lib/remote/Chunker.java

```diff
@@ -144,17 +144,17 @@ public void reset() throws IOException {
   }
 
   /**
-   * Seek to an offset, if necessary resetting or initializing
+   * Seek to an offset in the source stream.
    *
-   * <p>May close open resources in order to seek to an earlier offset.
+   * <p>May close and reopen resources in order to seek to an earlier offset.
    */
   public void seek(long toOffset) throws IOException {
-    if (toOffset < offset) {
+    if (initialized && toOffset >= offset && !compressed) {
+      ByteStreams.skipFully(data, toOffset - offset);
```

Comment:
Was just reviewing the release notes today for bazel 5.3.0 and came across this.

it looks like with this change, offset no longer updated here when skipFully is called. Just want to sanity check that this is the intentional behavior? (I am not super familiar with bazel internals and how seek is called, but just worried this would result in extra bytes discarded if offset is not updated)

Reply:
Thanks; this is a bug. It think it's unlikely to be triggered in practice, since seeking an initialized chunker forward is rare.

Reply:
got it, thanks for clarifying!